### ISSUE TITLE
Disable adding a feature when using cargo-add --optional

### ISSUE THREAD

**Issue by passcod** [rust-lang/cargo#13430]
### Problem

With #13071, `cargo add` started creating explicit features for new optional deps. While I understand the reasoning, this is frustrating to have to keep remembering to delete that line when I actively _want_ to use the newer behaviour that adding optional dependencies _doesn't_ create features.

### Proposed Solution

Some kind of opt-in switch or heuristic that will let me disable this behaviour.

**Comment by epage** [comment]
In addition to
- Pedagogical purposes
- Jump starting people who want to make the dependency a feature

We did this because it will be an error to do anything else in the next edition, see https://github.com/rust-lang/rfcs/pull/3491

Any switch will be short-lived.  Since this use case has the user editing the file either way, I don't think a switch would have been worth it.

However, like #13411, I wonder if we should add a Status message saying that we added the feature.  This way users are less likely to miss that it was done.
